src/main/java/mate/academy/Main.java

```diff
@@ -2,25 +2,28 @@
 
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 public class Main {
     public static void main(String[] args) {
         // Feel free to play with AsyncRequestProcessor in this main method if you want
-        ExecutorService executor = null; // Provide implementation that fits your needs
+        ExecutorService executor = Executors.newSingleThreadExecutor();
         AsyncRequestProcessor asyncRequestProcessor = new AsyncRequestProcessor(executor);
 
         // Simulating multiple concurrent requests
         String[] userIds = {"user1", "user2", "user3", "user1"}; // Note: "user1" is repeated
         CompletableFuture<?>[] futures = new CompletableFuture[userIds.length];
 
-        for (int i = 0; i < userIds.length; i++) {
-            String userId = userIds[i];
-            futures[i] = asyncRequestProcessor.processRequest(userId)
-                    .thenAccept(userData -> System.out.println("Processed: " + userData));
-        }
+        try {
+            for (int i = 0; i < userIds.length; i++) {
+                String userId = userIds[i];
+                futures[i] = asyncRequestProcessor.processRequest(userId)
+                        .thenAccept(userData -> System.out.println("Processed: " + userData));
+            }
+            CompletableFuture.allOf(futures).join();
+        } finally {
+            executor.shutdown();
```

Comment:
I do not think it was necessary in this case since it is just a main method which would exit from the application either way, but we can leave it